mdoc-js/src/main/scala/mdoc/modifiers/ScalajsWorker.scala

```diff
@@ -0,0 +1,264 @@
+    for {
+      irFiles <- pathIRContainer()
+      cache = irCache.newCache
+      clearableLinker = linker(config)
+      cached <- cache.cached(irFiles)
+    } yield ScalajsWorkerState(cached, clearableLinker)
+  }
+
+  def emptyConfig: _StandardConfig = new _StandardConfig(classes)
+
+  private final class FilteringClassLoader(parent: ClassLoader) extends ClassLoader(parent) {
+    private val parentPrefixes = List(
+      "java.",
+      "scala.",
+      "org.scalajs.linker.",
+      "org.scalajs.logging.",
+      "sun.reflect.",
+      "jdk.internal.reflect."
+    )
+
+    override def loadClass(name: String, resolve: Boolean): Class[_] = {
+      if (parentPrefixes.exists(name.startsWith _))
+        super.loadClass(name, resolve)
+      else
+        null
+    }
+  }
+
+  private val classloader =
+    new URLClassLoader(claspath, new FilteringClassLoader(getClass.getClassLoader()))
+
+  private implicit val classes: Classes = new Classes(classloader)
+
+  private def linker(config: _StandardConfig): _Linker =
+    new _StandardImpl(classes).linker(config)
+
+  private def irCache = new _StandardImpl(classes).irFileCache
+
+  private def pathIRContainer()(implicit ec: ExecutionContext): Future[Seq[_IRContainer]] = {
+    new _PathIRContainer(classes)
+      .fromClassPath(compilerClasspath.map(u => Paths.get(u.toURI)).toSeq)
+  }
+
+}
+
+class _StandardConfig(classes: Classes) { self =>
+  private val kls = classes.klassStandardConfig
+  private var impl =
+    classes.klassStandardConfig.getConstructor().newInstance().asInstanceOf[java.lang.Object]
+
+  def withModuleKind(kind: ModuleKind) = {
+    impl = kls
+      .getMethod("withModuleKind", classes.linkerClass("interface.ModuleKind"))
+      .invoke(impl, classes.linkerObject("interface." + kind.nm).getField("MODULE$").get(null))
+    self
+  }
+
+  def withOptimized(enable: Boolean) = {
+    val defaults = {
+      val kls = classes.linkerObject("interface.Semantics$")
+
+      val module = kls
+        .getField("MODULE$")
+        .get(null)
+
+      kls.getMethod("Defaults").invoke(module)
+    }
+
+    val klsSemantics = classes.linkerClass("interface.Semantics")
+
+    if (!enable)
+      impl = kls.getMethod("withSemantics", klsSemantics).invoke(impl, defaults)
+    else {
+      impl = kls
+        .getMethod("withSemantics", klsSemantics)
+        .invoke(impl, klsSemantics.getMethod("optimized").invoke(defaults))
+    }
+
+    self
+  }
+
+  private def flag(name: String, value: Boolean) = {
+    impl = kls
+      .getMethod("withSourceMap", classOf[Boolean])
+      .invoke(impl, java.lang.Boolean.valueOf(value))
+    self
+  }
+
+  def withSourceMap(enable: Boolean) =
+    flag("withSourceMap", enable)
+
+  def withBatchMode(enable: Boolean) =
+    flag("withBatchMode", enable)
+
+  def withClosureCompilerIfAvailable(enable: Boolean) =
+    flag("withClosureCompilerIfAvailable", enable)
+
+  def get = impl
+}
+
+class Classes(classloader: ClassLoader) {
```

Comment:
I know what you think: "Anton, this amount of reflection is awe inspiring and looks very robust and future proof".

But, counterpoint, it probably isn't.
A better solution is to create two (?) new modules like this:

```
                  Depends (%                                                    
┌────────────────┐"provided")┌───────────────┐ Implements  ┌───────────────────┐
│ Scalajs Linker │◀──────────│ ScalaJSWorker │────────────▶│ ScalaJSWorkerApi  │
└────────────────┘           └───────────────┘             └───────────────────┘
         ▲                           ▲                               ▲          
         │                           │                               │          
         └────────────────────────┬──┘                               │          
             ┌────────────────────┼─────────┐                        │          
             │Mdoc-js module      │         │                        │          
             │          ┌──────────────────┐│                        │          
             │          │     Isolated     ││                        │          
             │          │   classloader    │├───────Depends──────────┘          
             │          └──────────────────┘│                                   
             │                              │                                   
             └──────────────────────────────┘                                   
             1. Reflectively creates instance                                   
             of ScalaJSWorker                                                   
             2. Casts it as ScalaJSWorkerApi                                    
             3. Uses that                                                       
                                                                                
```

where ScalaJSWorker and ScalaJSWorkerAPI will become 2 new modules, published alongside existing mdoc-js module

Reply:
This was done in [e89e05f](https://github.com/scalameta/mdoc/pull/629/commits/e89e05f1679a6dc34cb8f5def944898297e6d217)